notifications/helm-repo-failure-20200918-.md

```diff
@@ -0,0 +1,131 @@
+# ACR Helm Repo Security Advisory
+
+|Date | Status |
+|-|-|
+| September 22, 2020| Active - testing solution |
+| September 18, 2020 | Identified |
+
+- The ACR engineering team has coded a fix that has been deployed for canary testing on September 22, 2020
+- Assuming testing goes as expected, following Azure safe deployment practices, regional deployments will start September 23, 2020
+- Deployments should complete by start of business, Monday September 28, 2020 - **_please check here for final confirmation_**
+
+## Does this Issue Apply to You
+
+- Using `helm repo` features with Azure Container Registry - yes
+- Using the helm client, version `v3.3.1` ***or lower*** - yes
+- Using **[helm 3 registry](https://helm.sh/docs/topics/registries/)** features that persist helm charts as OCI Artifacts - **this does NOT apply** to you
+
+## Issue Summary
+
+- ACR generates yaml content within the `chart.yaml` index to identify where the chart is stored within the content store
+  ```yaml
+  apiVersion: v1
+  entries:
+  wordpress:
+  - acrMetadata:
+      manifestDigest: sha256:08ef434162070bba4256414c80b001d15b7503ef2a1a4fa1f60bab174f80d4d7
+    appVersion: 5.1.0
+    created: "2019-03-06T16:59:25.8892193Z"
+  ```
+- To mitigate security concerns, unrelated to how ACR annotates `chart.yaml`, the helm client no longer supports additional content within the `chart.yaml` index file causing helm v3.3.2 or higher to fail.
+- ACR is actively rolling out a server side change to generate newly complaint content, expected to be completed by Monday September 28, 2020
+  ```yaml
+  apiVersion: v1
+  entries: wordpress:
+  - annotations
+    "azurecr.io/manifestDigest": sha256:08ef434162070bba4256414c80b001d15b7503ef2a1a4fa1f60bab174f80d4d7
+  appVersion: 5.1.0
+  created: "2019-03-06T16:59:25.8892193Z"
+  ```
+
+## Azure Container Registry User Guidance
+
+Evaluating the information provided to the [Helm security advisories](https://github.com/helm/helm/security/advisories):
+
+- [Repository index file allows for duplicates of the same chart entry](https://github.com/helm/helm/security/advisories/GHSA-jm56-5h66-w453)
+- [Sanitizing plugin names](https://github.com/helm/helm/security/advisories/GHSA-m54r-vrmv-hw33)
+- [plugin.yaml file allows for duplicate entries](https://github.com/helm/helm/security/advisories/GHSA-c52f-pq47-2r9j)
+
+We can provide some guidance to avoid customers being stuck between the proverbial rock and hard place while we rollout a server side change to regenerate the `chart.yaml` files.
+
+First, some scoping to this guidance:
+
+- [ACR is a private registry](https://aka.ms/acr), providing [private repository storage of Helm charts](https://aka.ms/acr/helm-repos)
+- ACR connectivity is secured through SSL connections, minimizing man in the middle attacks
```

Comment:
Would actually call out the exclusive list ponting to Azure Security Benchmarks since the scope of security is much larger than just SSL.